### PR TITLE
Document options

### PR DESCRIPTION
**Pull request by jcontonio**
* Adds Unpublish and Delete to the Document UI itself
* Adds other responsive issues for Document UI
* Removes flexbox from editor because of weird issue

### CODE REVIEW

File: web/src/client/js/components/Button/Button.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export { default as ButtonGroup } from './ButtonGroup'
```
Comment (dheniges) [inline web/src/client/js/components/Button/Button.js:5]: Is this file needed? If so, worth adding a comment

File: web/src/client/js/components/DocumentToolbar/DocumentToolbarComponent.js [hunk 2]
```diff
@@ -38,15 +48,37 @@ const DocumentToolbarComponent = ({
...
+          <DropdownComponent
+            align="right"
+            autoCollapse={true}
+            button={{
+              className: 'btn--small btn--with-icon',
+              icon: <CaretIcon fill="#fff" />
+            }}>
+            <DropdownItem
+              className="btn--small"
+              label="Unpublish document"
```
Comment (dheniges) [inline web/src/client/js/components/DocumentToolbar/DocumentToolbarComponent.js]: This button should get disabled if the doc hasn't been published, like it does in the document list menu

File: web/src/client/js/components/DocumentToolbar/DocumentToolbarComponent.js [hunk 3]
```diff
@@ -38,15 +48,37 @@ const DocumentToolbarComponent = ({
...
-          {!isPublishing && <PublishIcon fill="#ffffff" />}
-          <span className="label">Publish {publishString}</span>
-        </button>
+          small
+          title={publishButtonTitle()}
+        >
+          <button className="btn btn--small btn--with-icon" onClick={publishDocumentHandler}>
+            {isPublishing && <SpinnerComponent width="12" height="12" color="#ffffff" />}
+            {!isPublishing && <PublishIcon fill="#ffffff" />}
+            <span className="label">Publish {publishString}</span>
```
Comment (dheniges) [inline web/src/client/js/components/DocumentToolbar/DocumentToolbarComponent.js]: This publish button is no longer disabled after you publish a document and haven't made any changes

File: web/src/client/js/components/DocumentToolbar/DocumentToolbarContainer.js [hunk 4]
```diff
@@ -41,21 +50,48 @@ const DocumentToolbarContainer = ({ isCreating, isPublishing, publishDocument })
...
+    const confirmedLabel = `Yes, delete this document`
+    const confirmedAction = () => { deleteDocument(projectId, document.id, history) }
+    uiConfirm({ message, confirmedAction, confirmedLabel })
+  }
+
+  function unpublishDocumentHandler() {
+    const message = (
+      <>
+        <p>Unpublish the document <span className="highlight">{document.name}</span>?</p>
+        <p>If you are using this document in a live application, it will be removed.</p>
```
Comment (dheniges) [inline web/src/client/js/components/DocumentToolbar/DocumentToolbarContainer.js]: Probably worth moving this text/markup into a shared file so it's not repeated?

**Review by dheniges — CHANGES_REQUESTED** [review]
A few feedback items on this.

What do you think UX-wise? The "Unpublish" being an arrow menu option above the "Publish" button makes sense to me, but having the "Delete Document" in that same menu isn't as intuitive to me. Took me a few moments to notice the menu next to the button when I booted up this PR.

File: web/src/client/js/components/Button/Button.js (hunk 1 above)
Comment (jcontonio) [reply web/src/client/js/components/Button/Button.js:5]: Will add comment

**Review by dheniges — APPROVED** [review]
This feels nice and makes sense to me :+1:
